[#64069] Replace `.hidden-for-sighted` with Primer's `.sr-only`

myabc opened the pull request:
# Ticket

https://community.openproject.org/wp/64069

# What are you trying to accomplish?

Replace usages of own hand-rolled `hidden-for-sighted` class with Primer's `sr-only`.

See:
* Docs: https://primer.style/css/storybook/?path=/story/utilities-layout--screen-reader-only
* Source: https://github.com/primer/css/blob/main/src/utilities/visibility-display.scss#L96

## Screenshots

<img width="279" height="155" alt="Screenshot 2025-07-21 at 12 03 07" src="https://github.com/user-attachments/assets/7f5cee46-b604-45ef-8014-31690b8b8d39" />


# What approach did you choose and why?

The difference in implementation is as follows:

```diff
--- A/hidden-for-sighted.sass
+++ B/sr-only.sass
@@ -1,7 +1,10 @@
+.sr-only
-.hidden-for-sighted
   position: absolute
   width: 1px
   height: 1px
+  padding: 0
   overflow: hidden
+  clip-path: rect(0 0 0 0)
+  // Workaround for https://bugzilla.mozilla.org/show_bug.cgi?id=1241631
+  overflow-wrap: normal
+  border: 0
\ No newline at end of file
-  left: -10000px
-  top: 0
```

## Out-of-scope

Fixing existing Rubocop failures.

# Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [X] Tested major browsers (Chrome, Firefox, Edge, ...)

HDinger reviewed (changes requested):
I am not 100% sure whether this might cause some weird edge cases in some places. The `sr-only` class only positions absolutley, while our old class pushed it far to the left on top of that. So when you have a `sr-only` element within a relative element, you might see a weird pixel floating around.. Maybe this is however only a theoretical issue 🤷 

What is more important, is that this class is used in other repos as well (e.g. [here](https://github.com/opf/openproject-gitolite/blob/dev/app/views/repositories/_repository_header.html.erb#L73)). So if we want to remove it, we'd need to check the other repos as well.

myabc committed:
Replace .hidden-for-sighted with Primer's .sr-only

See (docs): https://primer.style/css/storybook/?path=/story/utilities-layout--screen-reader-only
See (source): https://github.com/primer/css/blob/main/src/utilities/visibility-display.scss#L96

myabc committed:
Fix features reliant on Capybara hidden heuristics

Capybara, when using #text or similar methods, typically only returns
text that is considered visible on the page. However, the definition of
"visible" depends on both the underlying driver and Capybara's
heuristics.

As such, Capybara + Cuprite consider the following rule in our
`.hidden-for-sighted` implementation's rule as "invisible":

    left: -10000px

However the Primer `.sr-only` implementation is considered "visible":

    clip-path: rect(0 0 0 0)

This commit updates the affected feature specs to account for this
difference in visibility detection. The tests now properly handle
screen reader only text that Capybara considers "visible" when using
Primer's `.sr-only` class instead of the custom `.hidden-for-sighted`
implementation.

HDinger commented:
The code changes here look good. Did you check other repositories that might depend on this class as I mentioned above?

myabc commented:
> Did you check other repositories that might depend on this class as I mentioned above?

@HDinger I checked all repos I have access to - and didn't find any usages.
